cgi-bin/DW/Captcha.pm

```diff
@@ -209,10 +209,14 @@ sub validate {
 
     my $err_ref = $opts{err_ref};
 
-    if ( $self->challenge ) {
-        return 1 if $self->_validate;
+                    # captcha type, page captcha appeared on
+    my $stat_tags = [ (ref $self)->name, "page:" . $self->page ];
```

Comment:
name can be “recaptcha", “textcaptcha"

page can be any of the keys in CAPTCHA_FOR in our config, currently:
“create”, “lost info”, “validate_openid”, “support_submit_anon”, “anon
post”, “autopost”, “comment_html_anon”, "comment_html_auth"


On Sat, Mar 7, 2015, at 10:28 PM, Mark Smith wrote:
> In cgi-bin/DW/Captcha.pm[1]:


> > @@ -209,10 +209,14 @@ sub validate {
>
>      my $err_ref = $opts{err_ref};
>
> -    if ( $self->challenge ) {
> -        return 1 if $self->_validate;
> +                    # captcha type, page captcha appeared on
> +    my $stat_tags = [ (ref $self)->name, "page:" . $self->page ];
>      What is name and page here? We can't insert completely arbitrary
>      tags, they have to fit some requirements -- basically
>      [a-z0-9_-:.] or something.


> — Reply to this email directly or view it on GitHub[2].

>



Links:

  1. https://github.com/dreamwidth/dw-free/pull/1285#discussion_r26002701
  2. https://github.com/dreamwidth/dw-free/pull/1285/files#r26002701